aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,394 @@
+public class MyController : Controller
+{
+    IHttpClientFactory _httpClientFactory;
+
+    public MyController(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public async Task<IActionResult> Index()
+    {
+        var client = _httpClientFactory.CreateClient();
+        var result = await client.GetStringAsync("http://myurl/");
+        return View();
+    }
+}
+```
+
+Using `HttpClientFactory` in this fashion is a great way to begin refactoring an existing app. It has no impact on the way `HttpClient` is used. In places where `HttpClient` instances are currently created, replace those with a call to `CreateClient`.
+
+## Named clients
+
+If an app requires multiple distinct uses of `HttpClient`, each with different configurations, an option is to use **named clients**. Any common configuration for a named `HttpClient` can be specified during registration.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient("github", c =>
+    {
+        c.BaseAddress = new Uri("https://api.github.com/");
+
+        c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // GitHub API versioning
+        c.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample");   // GitHub requires a user-agent
+    });
+    services.AddHttpClient();
+}
+```
+
+In the preceding code, `AddHttpClient` is called twice: once with the name "github" and once without. The GitHub-specific client has some default configuration applied&mdash;namely the base address and two headers required to work with the GitHub API.
+
+Each time `CreateClient` is called, a new instance of `HttpClient` is created and the configuration function is called.
+
+To consume a named client, a string parameter can be passed to `CreateClient`, specifying the name of the client to be created:
+
+```csharp
+public class MyController : Controller
+{
+    IHttpClientFactory _httpClientFactory;
+
+    public MyController(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public IActionResult Index()
+    {
+        var defaultClient = _httpClientFactory.CreateClient();
+        var gitHubClient = _httpClientFactory.CreateClient("github");
+        return View();
+    }
+}
+```
+
+In the preceding code, the `gitHubClient` variable has the `BaseAddress` and `DefaultRequestHeaders` set. The `defaultClient` variable doesn't. This provides the ability to have different configurations for different purposes. This could mean different configurations per endpoint/API, for example.
+
+`HttpClientFactory` creates, and caches, a single `HttpMessageHandler` per named client. Generally, a single TCP connection for each named client is used. When instantiating and disposing of a `HttpClient` manually, a TCP connection is used per instance. Creating multiple connections in this way can lead to socket exhaustion on the host.
```

Comment:
> a single TCP connection for each named client is used.

This isn't really correct. Each client handler has its own **pool** of connections, and the behavior is implementation/platform specific.

I think a correct statement to make would be to say that "generally the lifetime of connections used by a handler will not exceed the lifetime of the handler" (based on current implementations) 

Is this really the right time and place to introduce this kind of discussion?

Reply:
Agree, will move, reword this to the later section on lifetimes.